homeworks/HW1/src/CreateCourse.jsx

```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+import {Button, Form, FormGroup, Input, Col, Row, Container, Label} from 'reactstrap';
+import {SelectComponent} from './SelectComponent';
+
+export class CreateCourse extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            authors: [
+                {
+                    id: "1",
+                    name: "Spinoza",
+                    isSelected: false
+                },
+                {
+                    id: "2",
+                    name: "Nikolson",
+                    isSelected: false
+                },
+                {
+                    id: "3",
+                    name: "Dekker",
+                    isSelected: false
+                },
+                {
+                    id: "4",
+                    name: "Morningstar",
+                    isSelected: false
+                }
+            ]
+        }
+    }
+    
+    markItemAsSelectedOrRemoved = (elem, isSelected) => {
+        const name = elem.target.value;
+        const authors = this.state.authors;
+        authors.filter(item => item.name === name).map(function (item) {
```

Comment:
Please use reduce in this case. It's more preferable I guess.

Reply:
(filter operations = count of the array) + (map operations = count of the array after the filter)

reduce operations = count of the array